lib/elixir_boilerplate_graphql/schema.ex

```diff
@@ -3,6 +3,20 @@ defmodule ElixirBoilerplateGraphQL.Schema do
 
   alias ElixirBoilerplate.Repo
 
+  defmodule Introspection do
+    @disable_introspection Application.compile_env(:elixir_boilerplate, ElixirBoilerplateGraphQL)[:enable_introspection]
+
+    def pipeline(pipeline) do
+      if @disable_introspection do
+        pipeline
+      else
+        Absinthe.Pipeline.without(pipeline, Absinthe.Phase.Schema.Introspection)
+      end
+    end
+  end
```

Comment:
Since it’s a compile time module attributes, we can save the conditional at runtime. Also I think the condition is not correct?

```suggestion

    if @disable_introspection do
       def pipeline(pipeline), do: Absinthe.Pipeline.without(pipeline, Absinthe.Phase.Schema.Introspection)
    else
      def pipeline(pipeline), do: pipeline
    end
```

Reply:
Oh, good catch! The condition was right, but the naming was wrong indeed…

```elixir
@disable_introspection Application.compile_env(…)[:enable_introspection]
```

🤦‍♂️